server/lib/actors/fetcher.go

```diff
@@ -0,0 +1,81 @@
+package actors
+
+import (
+	"fmt"
+	"database/sql"
+	"time"
+
+	"github.com/umegaya/channer/server/lib/models"
+	proto "github.com/umegaya/channer/server/proto"
+
+	"github.com/umegaya/yue"
+)
+
+var FLAME_THRESHOLD float64 = 0.1
+var FLAME_VOTE_THRESHOLD uint32 = 10
+
+func topicFetcher(btyp proto.TopicListRequest_BucketType, start, end time.Time, locale string) ([]FetchResult, error) {
+	entries := make([]FetchResult, DEFAULT_HOTBUCKET_SIZE)
+	dbm := models.DBM()
+	var rows *sql.Rows
+	var err error
+	locale_clause := ""
+	if len(locale) > 0 {
+		locale_clause = fmt.Sprintf(" and locale = '%s'", locale)
+	}
+	switch btyp {
+	case proto.TopicListRequest_Rising:
+		where_clause := fmt.Sprintf("created >= %v and created < %v%s", yue.UUIDAt(start), yue.UUIDAt(end), locale_clause)
+		rows, err = dbm.Query(
+			dbm.Stmt(
+				`select id,max(parent) as channel,sum(param) as point,count(persona) as vote from %s.reactions where %s and type = $1 group by id`, 
+			where_clause), 
+		proto.Model_Reaction_Topic_Vote)
+	case proto.TopicListRequest_Hot:
+		where_clause := fmt.Sprintf("id >= %v and id < %v%s", yue.UUIDAt(start), yue.UUIDAt(end), locale_clause)
+		rows, err = dbm.Query(
+			dbm.Stmt(
+				`select id,channel,point,vote from %s.topics where %s order by point desc limit %v`, 
+			where_clause, len(entries)))
+	case proto.TopicListRequest_Flame:
+		/*
+		u: upvote, d: downvote, v: vote, p: point
+		u + d = v
+		u - d = p
+
+		u = (v + p) / 2
+		d = (v - p) / 2
+
+		u / v => (1 + p / v) / 2
+
+		abs(u / v - 0.5) = abs(p / (2 * v)) < threshold => abs(p / v) < 2 * threshold
+		*/
+		where_clause := fmt.Sprintf("id >= %v and id < %v%s", yue.UUIDAt(start), yue.UUIDAt(end), locale_clause)
+		rows, err = dbm.Query(
```

Comment:
__[gometalinter]__
* [aligncheck] invalid operation: dbm (variable of type *github.com/umegaya/channer/server/lib/models.Database) has no field or method Query